software/contrib/pong.py

```diff
@@ -0,0 +1,230 @@
+        return self.handle_score()
+
+    def handle_bounce(self):
+        if self.y < 0:
+            wall_sound.trigger()
+            self.y *= -1
+            self.dy *= -1
+        elif self.y > oled.height-1:
+            wall_sound.trigger()
+            self.y = oled.height - (self.y - oled.height)
+            self.dy *= -1
+
+    def check_paddle_collision(self, paddle_x, paddle_y, paddle_height, dir: Paddle):
+        if self.x < paddle_x and dir == Paddle.RIGHT:
+            return False
+        if self.x > paddle_x and dir == Paddle.LEFT:
+            return False
+        if self.y < paddle_y:
+            return False
+        if self.y > paddle_y + paddle_height:
+            return False
+        return True
+
+    def handle_paddle_collision(self, paddle_x, paddle_y, paddle_height, dir):
+        if self.check_paddle_collision(paddle_x, paddle_y, paddle_height, dir):
+            paddle_sound.trigger()
+            self.x = paddle_x - (self.x - paddle_x)
+            self.dx *= -1
+            self.dx += math.copysign(0.8, self.dx)
+
+    def check_score(self) -> Paddle:
+        if self.x < 0:
+            return Paddle.RIGHT
+        elif self.x > oled.width:
+            return Paddle.LEFT
+        else:
+            return Paddle.NONE
+
+    def handle_score(self):
+        score = self.check_score()
+        if score != 0:
+            score_sound.trigger()
+            self.reset()
+            return score
+        else:
+            return 0
+
+    def reset(self):
+        self.x = oled.width // 2
+        self.y = oled.height // 2
+        self.randomize_direction()
+
+    def draw(self):
+        oled.rect(int(self.x), int(self.y), 1, 1, 1)
+
+
+class Pong(EuroPiScript):
+    def __init__(self):
+        super().__init__()
+        
+        # used to avoid screen burn-in
+        self.line_shift = False
+
+        self.ball = Ball()
+        self.ball.reset()
+        self.paddle_height = oled.height // 4
+        self.pad_pos = [0, 0]
+        self.score = [0, 0]
+
+    def draw_paddle(self, x, y):
+        oled.vline(x, y, self.paddle_height, 1)
+
+    def update_paddle_positions(self):
+        self.pad_pos[0] = k1.read_position(oled.height - self.paddle_height)
+        self.pad_pos[1] = k2.read_position(oled.height - self.paddle_height)
+
+    def draw_midline(self):
+        for i in range(self.line_shift * 2, oled.height, 4):
+            oled.vline(oled.width // 2, i, 2, 1)
+
+    def update_paddles(self):
+        self.update_paddle_positions()
+        self.draw_paddle(1, self.pad_pos[0])
+        self.draw_paddle(oled.width - 1, self.pad_pos[1])
+
+    def draw(self):
+        oled.fill(0)
+        self.draw_midline()
+        self.update_paddles()
+        scored = self.ball.update(1, self.pad_pos[0], oled.width-1, self.pad_pos[1], self.paddle_height)
+        self.ball.draw()
+        if scored != Paddle.NONE:
+            self.score[scored-1] += 1
```

Comment:
[required] There's a bug somewhere in the score code (probably in the calculation of this index?) which causes the left player's score to increment for every score and never the right's.